### PR TITLE
Complete day 6 tasks and tests

### PR DESCRIPTION
**Pull request by csmangum**
Add Day 6 metrics to track engine performance, perception selectivity, and prediction accuracy.

This PR introduces new instrumentation in the engine to record the time steps for unit state transitions (REQUESTED, ACTIVE, TRUE, CONFIRMED) and count terminal requests. A new `metrics` module provides utilities to calculate these performance indicators, such as steps to first TRUE/CONFIRM and total/per-terminal SUR requests, along with a `binary_precision_recall` function for general prediction evaluation. These metrics enable detailed analysis of the network's efficiency and active perception.

---
<a href="https://cursor.com/background-agent?bcId=bc-997b679a-1f4e-4172-a995-e07813e3f6d0">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-cursor-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-cursor-light.svg">
    Open in Cursor
  </picture>
</a>
<a href="https://cursor.com/agents?id=bc-997b679a-1f4e-4172-a995-e07813e3f6d0">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-web-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-web-light.svg">
    Open in Web
  </picture>
</a>

### CODE REVIEW

**Comment by csmangum** [comment]
This pull request introduces a comprehensive runtime metrics system ("Day 6 metrics") for the active perception engine, enabling measurement of efficiency, timing, and event tracking. It adds a new `metrics.py` module with binary classification helpers and engine statistics accessors, instruments the engine to record detailed metrics, exposes these via the API, and documents their usage. New unit tests and lightweight test runner support are included to ensure correctness.

**Metrics system and engine instrumentation:**

* The `Engine` class now records runtime statistics such as total SUR requests to TERMINALs, per-terminal request counts, and timing for key state transitions (`first_request_step`, `first_active_step`, `first_true_step`, `first_confirm_step`). These are reset on `Engine.reset()` and included in `Engine.snapshot()` output. [[1]](diffhunk://#diff-ad7da5324f79594c8b3381594be3e4c464cddbd0ca1e0a61503de987a835e4e1R47-R57) [[2]](diffhunk://#diff-ad7da5324f79594c8b3381594be3e4c464cddbd0ca1e0a61503de987a835e4e1R73-R82) [[3]](diffhunk://#diff-ad7da5324f79594c8b3381594be3e4c464cddbd0ca1e0a61503de987a835e4e1R267-R278) [[4]](diffhunk://#diff-ad7da5324f79594c8b3381594be3e4c464cddbd0ca1e0a61503de987a835e4e1L271-R320) [[5]](diffhunk://#diff-ad7da5324f79594c8b3381594be3e4c464cddbd0ca1e0a61503de987a835e4e1R341-R342) [[6]](diffhunk://#diff-ad7da5324f79594c8b3381594be3e4c464cddbd0ca1e0a61503de987a835e4e1R429)

* SUR requests to terminal units are now counted exactly once per parent activation episode, preventing over-counting and enabling accurate efficiency measurement.

**Metrics utilities and API:**

* Added new module `recon_core/metrics.py` providing:
  - `binary_precision_recall` for evaluating binary predictions,
  - `steps_to_first_true`, `steps_to_first_confirm` for event timing,
  - `total_terminal_requests`, `terminal_request_counts_by_id` for efficiency stats.
  These helpers are exposed at the package level and documented for user access. [[1]](diffhunk://#diff-8b028861aca932913c9910488ad79d5747917a3ebe18feef88b87e26112b9f56R1-R76) [[2]](diffhunk://#diff-2f2313c105dec998664376a60f24ca5ee735da76c6e93a7553a2cb34a3544878R30-R46)

**Documentation and usage examples:**

* The `README.md` and `RECON_DOCUMENTATION.md` now describe the new metrics system, available helpers, and provide code examples for evaluating predictions and inspecting engine statistics. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R45) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L80-R117) [[3]](diffhunk://#diff-2674c2547ec664e240007b36d5a1230c5ff0e7e0ea44b7134312bcf9c0225842R148-R185)

**Testing and test runner updates:**

* Added `tests/test_metrics.py` with unit tests for all new metrics utilities and engine counters, including edge cases and reset behavior.
* The lightweight `run_tests.py` runner is updated to include Day 6 metrics tests, ensuring coverage even in minimal environments.

Overall, these changes provide robust instrumentation and evaluation tools for active perception experiments, with thorough documentation and testing.

**References:** [[1]](diffhunk://#diff-ad7da5324f79594c8b3381594be3e4c464cddbd0ca1e0a61503de987a835e4e1R47-R57) [[2]](diffhunk://#diff-ad7da5324f79594c8b3381594be3e4c464cddbd0ca1e0a61503de987a835e4e1R73-R82) [[3]](diffhunk://#diff-ad7da5324f79594c8b3381594be3e4c464cddbd0ca1e0a61503de987a835e4e1R267-R278) [[4]](diffhunk://#diff-ad7da5324f79594c8b3381594be3e4c464cddbd0ca1e0a61503de987a835e4e1L271-R320) [[5]](diffhunk://#diff-ad7da5324f79594c8b3381594be3e4c464cddbd0ca1e0a61503de987a835e4e1R341-R342) [[6]](diffhunk://#diff-ad7da5324f79594c8b3381594be3e4c464cddbd0ca1e0a61503de987a835e4e1R429) [[7]](diffhunk://#diff-8b028861aca932913c9910488ad79d5747917a3ebe18feef88b87e26112b9f56R1-R76) [[8]](diffhunk://#diff-2f2313c105dec998664376a60f24ca5ee735da76c6e93a7553a2cb34a3544878R30-R46) [[9]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R45) [[10]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L80-R117) [[11]](diffhunk://#diff-2674c2547ec664e240007b36d5a1230c5ff0e7e0ea44b7134312bcf9c0225842R148-R185) [[12]](diffhunk://#diff-c013839ae9f8d11da7ba9ed964de9d4b9ef733c1dd3d1a5487f32a5b2062e1cfR1-R109) [[13]](diffhunk://#diff-5d20bf301d1d889d7ec73948673520e35c59a2f06b78ea646f0d7f0348ff6669R335-R402)

**Comment by csmangum** [comment]
bugbot run

File: recon_core/engine.py [hunk 1]
```diff
@@ -268,15 +294,30 @@ def _update_states(self, delta: Dict[str, float]):
                 if u.state == State.INACTIVE and u.a > 0.1:
                     if u.state != State.REQUESTED:
                         u.state = State.REQUESTED
-                        # Send REQUEST to children via SUR links
-                        for child_id in self.g.sur_children(u.id):
-                            u.outbox.append((child_id, Message.REQUEST))
+                        if uid not in self.stats["first_request_step"]:
+                            self.stats["first_request_step"][uid] = self.t
```
[the remark this answers is not in the document]
Comment (csmangum) [reply recon_core/engine.py:298]: @cursor make the fix
